src/main/kotlin/com/hibob/ex2-loops.kt

```diff
@@ -0,0 +1,40 @@
+package com.hibob
+
+fun isContainsSpecialCharacterExcludingUnderscore(s: String): Boolean {
+    val regex = "[^a-zA-Z0-9_]".toRegex()
+    return regex.containsMatchIn(s)
+}
+
+fun isContainOnlyNumbers(s: String): Boolean {
+    var numberOnly = true
+    for (char in s) {
+        if (char in '0'..'9') continue else numberOnly = false
```

Comment:
It's not good practice to use `continue`. If you negate the `if` term, the `continue` becomes redundant.
```
if (char not in '0'..'9') numberOnly = false
```